docsite/rst/developing_program_flow_modules.rst

```diff
@@ -0,0 +1,392 @@
+.. _flow_modules:
+
+=======
+Modules
+=======
+
+This is an in-depth dive into understanding Ansible's program flow to execute
+modules.  It will be of use to people working on the portions of the Core
+Ansible Engine that execute a module, it may be of interest to people writing
+Ansible Modules, and people simply wanting to use Ansible Modules will likely
+not find anything helpful here.
+
+.. _flow_types_of_modules:
+
+Types of Modules
+================
+
+Ansible supports several different types of modules in its code base.  Some of
+these are for backwards compatibility and others to enable flexibility.
+
+.. _flow_action_plugins:
+
+Action Plugins
+--------------
+
+Action Plugins look like modules to end users who are writing :term:`playbooks` but
+they're distinct entities for the purposes of this paper.  Action Plugins
+always execute on the controller. They sometimes are able to do all work there
```

Comment:
combine last two sentences:
Action Plugins always execute on the controller and are sometimes able to do all work there (...).